tests/data_properties_v3_conversion/genius_zone_data_state_test.py

```diff
@@ -0,0 +1,248 @@
+        genius_zone = GeniusZone(self._device_id, self.raw_json, self.hub)
+        self.assertFalse(genius_zone.data['_state']['bOutRequestHeat'])
+
+    def test_when_bOutRequestHeat_is_true_then_state_bOutRequestHeat_true(self):
+        "Check that the bOutRequestHeat is correctly set to true"
+
+        self.raw_json["bOutRequestHeat"] = 1
+        genius_zone = GeniusZone(self._device_id, self.raw_json, self.hub)
+        self.assertTrue(genius_zone.data['_state']['bOutRequestHeat'])
+
+    def test_when_iMode_set_then_state_mode_is_set_correctly(self):
+        "Check that the mode is set on the class"
+
+        for zone_mode, zone_mode_text in IMODE_TO_MODE.items():
+            with self.subTest(zone_mode=zone_mode, zone_mode_text=zone_mode_text):
+                self.raw_json["iMode"] = zone_mode
+                self.raw_json["zoneSubType"] = 1
+                genius_zone = GeniusZone(self._device_id, self.raw_json, self.hub)
+
+                self.assertEqual(genius_zone.data['mode'], zone_mode_text)
+
+    def test_when_iType_should_set_temperature_state_temperature_set_correctly(self):
+        "Check that the temperature is set for certain values of iType"
+
+        temperature = 20.0
+        setpoint = 21.0
+
+        test_values = [
+            ZONE_TYPE.ControlSP,
+            ZONE_TYPE.TPI,
+            ZONE_TYPE.Manager
+        ]
+        self.raw_json["fPV"] = temperature
+        self.raw_json["fSP"] = setpoint
+
+        for zone_type in test_values:
+            with self.subTest(zone_type=zone_type):
+                self.raw_json["iType"] = zone_type
+                genius_zone = GeniusZone(self._device_id, self.raw_json, self.hub)
+
+                self.assertEqual(genius_zone.data['temperature'], temperature)
+
+    def test_when_iType_should_not_set_temperature_state_temperature_not_set(self):
+        "Check that the temperature is not set for certain values of iType"
+
+        temperature = 20.0
+        setpoint = 21.0
+
+        test_values = [
+            ZONE_TYPE.OnOffTimer,
+            ZONE_TYPE.ControlOnOffPID,
+            ZONE_TYPE.Surrogate
+        ]
+        self.raw_json["fPV"] = temperature
+        self.raw_json["fSP"] = setpoint
+
+        for zone_type in test_values:
+            with self.subTest(zone_type=zone_type):
+                self.raw_json["iType"] = zone_type
+                genius_zone = GeniusZone(self._device_id, self.raw_json, self.hub)
+
+                self.assertFalse('temperature' in genius_zone.data)
+
+    def test_when_iType_should_set_setpoint_state_setpoint_set_correctly(self):
+        "Check that the setpoint is set for certain values of iType"
+
+        temperature = 20.0
+        setpoint = 21.0
+
+        test_values = [
+            ZONE_TYPE.ControlSP,
+            ZONE_TYPE.TPI
+        ]
+        self.raw_json["fPV"] = temperature
+        self.raw_json["fSP"] = setpoint
+
+        for zone_type in test_values:
```

Comment:
in fact, the Python tradition might be:
```python
for zone_type in (ZONE_TYPE.ControlSP, ZONE_TYPE.TPI):
```

Reply:
If possible, I would prefer to leave this as is, as it follows the Arrange / Act / Assert pattern for unit tests, which I do think provides value in the clarity of what the test is doing.

Reply:
OK, I understand - any chance you could put in comments for this, otherwise it's possible it might get refactored!

Reply:
I would rather not, if that's alright, purely on the grounds that I'd need to add comments to, at the very least, every unit test file.

However, you've made me realise that I haven't added anything about unit tests to the readme for this project, so I've updated that to include the information about how I've written the unit tests here.

Is that an acceptable solution?

Reply:
Sure. But could you just post one here - in the comments - so I can see what you mean. Is it:
```python
        # arrange
        temperature = 20.0

        test_values = (
            ZONE_TYPE.ControlSP,
            ZONE_TYPE.TPI,
            ZONE_TYPE.Manager
        )

        self.raw_json["fPV"] = temperature
        self.raw_json["fSP"] = 21.0

        # act
        for zone_type in test_values:
            with self.subTest(zone_type=zone_type):
                self.raw_json["iType"] = zone_type
                genius_zone = GeniusZone(self._device_id, self.raw_json, self.hub)

                self.assertEqual(genius_zone.data["temperature"], temperature)  # assert
```

Reply:
Funnily enough, comments were the one thing that I was making sure I didn't use for this! This stems from my last job, where someone put in comments for each section, and then people ended up putting the comments in so many tests!

So, I've added a few examples to the read me file, is this acceptable?

Oh, I also realised that I'd forgotten to put in the command to run the tests, as there's a specific parameter needed to get the tests picked up.

Reply:
I tell you what - how about just 1-2 sentences in the repo README, and pull the rest out a README under /tests?

Reply:
That makes sense.

Reply:
Done.